util/settings/settings.go

```diff
@@ -1743,7 +1743,11 @@ func (a *ArgoCDSettings) OAuth2ClientID() string {
 func (a *ArgoCDSettings) OAuth2AllowedAudiences() []string {
 	if config := a.oidcConfig(); config != nil {
 		if len(config.AllowedAudiences) == 0 {
-			return []string{config.ClientID}
+			allowedAudiences := []string{config.ClientID}
+			if config.CLIClientID != "" {
+				allowedAudiences = append(allowedAudiences, config.CLIClientID)
+			}
```

Comment:
Thanks for the fix! We should also update docs to mention 1) that this is an allowed audience by default, if configured and 2) if someone manually sets `allowedAudiences`, they must explicitly add this audience, if wanted.